sdk/java-sdk-protobuf/src/main/scala/kalix/javasdk/impl/eventsourcedentity/EventSourcedEntitiesImpl.scala

```diff
@@ -234,7 +236,12 @@ final class EventSourcedEntitiesImpl(
                         serializedSnapshot,
                         delete))))
             }
-          } finally { span.foreach(_.end()) }
+          } finally {
+            span.foreach { s =>
+              mdc.remove(Telemetry.TRACE_ID)
```

Comment:
Removing trace id in a finally block on the same thread is good, but needs to remove from the actual MDC rather than the copied map.

Reply:
done